src/pages/home/Home.tsx

```diff
@@ -4,23 +4,37 @@ import trashIcon from "../../assets/trash-icon.webp";
 import activateIcon from "../../assets/activate-icon.svg";
 import editIcon from "../../assets/edit-icon.png";
 import { useNavigate } from "react-router";
-import { useGetUserStudentsQuery } from "../../app/services/UserService";
 import useAuthentication from "../../hooks/useAuthentication";
 import { useDeleteStudentMutation } from "../../app/services/StudentService";
 import { ConfirmDialog } from "../../components/confirm_dialog/ConfirmDialog";
 import { useState } from "react";
+import { useGetUserStudentsQuery } from "../../app/services/UserService";
+import Filter from "../../components/filter/Filter";
+import { skipToken } from "@reduxjs/toolkit/query";
 
 function Home() {
   const { userId } = useAuthentication();
-  const { data: students } = useGetUserStudentsQuery(userId!);
   const [deleteStudent] = useDeleteStudentMutation();
   const navigate = useNavigate();
+  const [filter, setFilter] = useState("");
+  const [debouncedFilter, setDebouncedFilter] = useState("");
```

Comment:
Siguiendo el comentario anterior, este debouncedFilter podrías quitarlo y dejar solamente filter


```suggestion
  const [filter, setFilter] = useState("");
```